subprojects/plugins/src/test/groovy/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponentTest.groovy

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.component.internal
+
+
+import org.gradle.api.internal.tasks.JvmConstants
+import org.gradle.api.plugins.JavaBasePlugin
+import org.gradle.api.plugins.JavaPluginExtension
+import org.gradle.api.tasks.SourceSet
+import org.gradle.test.fixtures.AbstractProjectBuilderSpec
+
+/**
+ * Tests {@link DefaultJvmSoftwareComponent}.
+ */
+class DefaultJvmSoftwareComponentTest extends AbstractProjectBuilderSpec {
+
+    def "configures the project"() {
+        when:
+        project.plugins.apply(JavaBasePlugin)
+        def ext = project.getExtensions().getByType(JavaPluginExtension.class)
```

Comment:
🧐 Perhaps a bit nitpicky, but I think there would be some value in asserting the expected project elements you test for below do *not* exist at this point.  Refactoring moving stuff into the `JavaBasePlugin` seems a likely failure mode that this test would not detect.